liferay-echo/liferay-portal
# LPS-179772

Motivation
=======
Categories cannot be seen when editing web content if the user has a site role. If the user has a regular role the categories can be seen.
This can only be reproduced with the customer database that has an odd set of categories and permissions.
[LPS-179772](https://issues.liferay.com/browse/LPS-179772)

Proposed Solution
========
Allow categories to be included in the JSON if the current category can be seen with the current role. Before the change, if one category cannot be seen then all the categories will not be listed.

Steps to Verify
========
(This could only be verified with the customer database where site "detran" exist, I tried to create different sites with different categories in a clean environment and could not reproduce the issue)
1 - Create a regular user with no role assigned to him.
2 - Creating Roles [Regular Role] - In the browser logged in as admin: Control Panel > Roles > Regular Role > Add New Role > Type: Regular > Save.
After creating the Regular Role, define the permissions in: On the left menu, Choose Site and Asset Library Administration > Content & Data > Web Content > Check all options > Save.
Next, Choose Site and Asset Library Administration > Categorization > Categories > Check all options > Save. Notice here that totaled 62 permissions for this user.
3 - Creating Roles [Site Role] - Which will be inherited by the subscribed user - In the browser logged in as admin: Control Panel > Roles > Site Roles > Add New Role > Type: Site > Save
4 - After creating the Site Role, define the same permissions that were defined for the Regular Role in the actions above. Both users with the same 62 permissions
5 -  On the site http://localhost:8080/web/detran, access the Site settings: Page Edit Menu > Configuration > Site Configuration > Advanced - Here make sure that the Detran site is configured as the default site role the role that was previously created

Test 1: In the browser logged in as admin, go to Control Panel > Users and Organizations > Test User created > Roles > Regular Roles > Add the Regular Role that was created to the new user

6 - Return to the browser logged in with the test user and when trying to create a Web Content notice that the vocabularies and categories are visible and can be managed.

Test 2: In the browser logged in as admin, access Control Panel > Users and Organizations > User Test created > Remove the regular role linked to this user > Save > Go to Memberships > Add the DETRAN site to this user's registrations > Save. Return to the Roles tab and notice that the user automatically inherited the site role that was linked to the site he was subscribed to.

Return to the browser logged in with the test user and when trying to create a Web Content notice that the vocabularies and categories are not visible and cannot be managed

**Expected Behavior:**

User should view and manage categories in both roles since both have the same permissions

**Actual Behavior:**

Only user as regular role as administrator can view - The site's inherited role does not fulfill its function.
